lib/experimental/script-modules.php

```diff
@@ -201,6 +201,43 @@ function gutenberg_dequeue_module( $module_identifier ) {
 	wp_script_modules()->dequeue( $module_identifier );
 }
 
+/**
+ * Prints HTML for the a11y Script Module.
+ *
+ * a11y relies on some DOM elements to use as ARIA live regions.
+ * Ideally, these elements are part of the initial HTML of the page
+ * so that accessibility tools can find them and observe updates.
+ */
+function gutenberg_a11y_script_module_html() {
+	$a11y_module_available = false;
+
+	$get_marked_for_enqueue = new ReflectionMethod( 'WP_Script_Modules', 'get_marked_for_enqueue' );
+	$get_marked_for_enqueue->setAccessible( true );
+	$get_import_map = new ReflectionMethod( 'WP_Script_Modules', 'get_import_map' );
+	$get_import_map->setAccessible( true );
```

Comment:
Script Modules is going to need a function to ask "is the script module available" (enqueued or in the import map). There isn't another good way to access this now. This same pattern already exists in Gutenberg:

https://github.com/WordPress/gutenberg/blob/c0e3ad898f67792d48c5e3b45f14b7a2c189722b/lib/compat/wordpress-6.7/script-modules.php#L15-L26

Reply:
It's about time to take action in WordPress core and expose the missing functionality for extenders:

- `get_import_map` should become public, it contains `imports` field only, so it can get extended but it's enough for now as it's exactly the same shape that gets printed in HTML
- add public method `is_marked_for_enqueue`
- add public method `is_marked_for_import`

Bonus points for adding a filter that gets called before returning the value from `get_import_map` call so 3rd party plugin could apply their changes.

Reply:
I propagated my feedback to WordPress Trac under related ticket: https://core.trac.wordpress.org/ticket/60597#comment:15.